### PR TITLE
Improve recipes and excluded them from C# codebase if needed

### PR DESCRIPTION
**Pull request by jevanlingen**
## What's changed?
Excluded CSharp for some recipes + made some recipes a little bit more modern

## What's your motivation?
Some recipes don't work for C#.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Review by timtebeek — APPROVED** [review]
Looks good; what part of this is still in draft? Or can we already merge this increment?

File: src/main/java/org/openrewrite/staticanalysis/StringLiteralEquality.java [hunk 1]
```diff
@@ -110,23 +116,6 @@ private J.Unary asNegatedUnary(J.MethodInvocation mi) {
...
-                    } else if (binary.getOperator() == J.Binary.Type.NotEqual) {
-                        after = asNegatedUnary(asEqualsMethodInvocation(binary)).withPrefix(binary.getPrefix());
-                    }
-                    if (after != null) {
-                        doAfterVisit(new EqualsAvoidsNull().getVisitor());
-                        return after;
-                    }
-                }
-                return super.visitBinary(binary, ctx);
-            }
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/staticanalysis/StringLiteralEquality.java:129]: Did you check which commit had added this override? That might have missed adding a test. Not sure why it was removed now?

Comment (jevanlingen) [reply src/main/java/org/openrewrite/staticanalysis/StringLiteralEquality.java:129]: I did not remove it, I just moved the method upwards (so public method first, then the private helper methods).

Comment (timtebeek) [reply src/main/java/org/openrewrite/staticanalysis/StringLiteralEquality.java:129]: I can see it's _old_, but when running with coverage it does appear to be hit.
image

Comment (timtebeek) [reply src/main/java/org/openrewrite/staticanalysis/StringLiteralEquality.java:129]: Ah it's merely moved up; that's at times hard to spot in a diff.

**Pull request merged**